hack/lib/test.sh

```diff
@@ -223,3 +228,20 @@ kube::test::if_has_string() {
     return 1
   fi
 }
+
+# Returns true if the required resource is part of supported resources.
+# Expects env vars:
+#   SUPPORTED_RESOURCES: Array of all resources supported by the apiserver. (*)
+#   means it supports all resources.
+#   $1: Name of the resource to be tested.
+kube::test::if_supports_resource() {
+  SUPPORTED_RESOURCES=${SUPPORTED_RESOURCES:-""}
+  REQUIRED_RESOURCE=${1:-""}
+
+  for r in "${SUPPORTED_RESOURCES[@]}"; do
+    if [[ "${r}" == "*" || "${r}" == "${REQUIRED_RESOURCE}" ]]; then
+      return 0
```

Comment:
Ignore this. I am not sure what I was thinking. Bash doesn't allow returning non-numeric exit codes.